controllers/admin/AdminCarriersController.php

```diff
@@ -649,6 +649,23 @@ public function getList($id_lang, $order_by = null, $order_way = null, $start =
                 $this->_list[$key]['name'] = Carrier::getCarrierNameFromShopName();
             }
         }
+
+        // Add specific sort for naming
+        if ($this->_orderBy == 'name') {
+            $orderWay = $this->_orderWay;
+            usort($this->_list, function (array $a, array $b) use ($orderWay) {
+                $aName = strtolower($a['name']);
+                $bName = strtolower($b['name']);
+                if ($aName == $bName) {
+                    return 0;
+                }
+                if ($orderWay == 'ASC') {
+                    return ($aName < $bName) ? -1 : 1;
+                } else {
+                    return ($aName > $bName) ? -1 : 1;
+                }
+            });
```

Comment:
Could you please externalise the method instead of using anonymous one.
It also need a little refactoring like testing the key 'name' exists, and remove useless else at the end :)

Reply:
Fixed